src/Controllers/LoginSessionController.php

```diff
@@ -29,42 +33,48 @@ class LoginSessionController extends Controller
      */
     public function remove(HTTPRequest $request): HTTPResponse
     {
-        return $this->removeLoginSession($request);
-    }
-
-    private function removeLoginSession(HTTPRequest $request): HTTPResponse
-    {
-        $failureMessage = _t(__CLASS__ . '.REMOVE_FAILURE', 'Something went wrong.');
-        try {
-            // Ensure CSRF protection
-            if (!SecurityToken::inst()->checkRequest($request)) {
-                return $this->jsonResponse([
+        if (empty(Security::getCurrentUser())) {
+            return $this->jsonResponse(
+                [
                     'success' => false,
-                    'message' => $failureMessage
-                ]);
-            }
+                    'message' => _t(
+                        __CLASS__ . '.YOUR_SESSION_HAS_EXPIRED',
+                        'Your session has expired'
```

Comment:
I'm adding a check to make sure the user is logged in when they try to invalidated a session.

@silverstripeux "Your session has expired" will be displayed to the user when they try to invalidate a session while their current session has expired. Let me know if you want a different wording for this.

Reply:
Here's what it looks like. 
![image](https://user-images.githubusercontent.com/1168676/121979711-ce269180-cdde-11eb-8fe6-119b71043733.png)